inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp

```diff
@@ -203,18 +206,42 @@ void Engine::AddExtension(InferenceEngine::IExtensionPtr extension) {
 }
 
 void Engine::QueryNetwork(const ICNNNetwork& network, const std::map<std::string, std::string>& config, QueryNetworkResult& res) const {
-    details::CNNNetworkIterator i(&network);
-    while (i != details::CNNNetworkIterator()) {
-        try {
-            mkldnn::engine eng(mkldnn::engine(mkldnn::engine::kind::cpu, 0));
-            MKLDNNWeightsSharing::Ptr fake_w_cache;
-
-            // if we can create and have not thrown exception, then layer is supported
-            std::unique_ptr <MKLDNNNode>(MKLDNNNode::CreateNode(*i, eng, extensionManager, fake_w_cache));
-            res.supportedLayersMap.insert({ (*i)->name, GetName() });
-        } catch (InferenceEngine::details::InferenceEngineException&) {
+    MKLDNNWeightsSharing::Ptr fake_w_cache;
+    auto function = network.getFunction();
+    if (function != nullptr) {
+        std::unordered_set<std::string> originalOps;
+        for (auto&& node : function->get_ops()) {
+            if (!node->is_constant() && !node->is_parameter() && !node->is_output()) {
+                originalOps.emplace(node->get_friendly_name());
+            }
+        }
+        auto clonedNetwork = cloneNetwork(network);
+        clonedNetwork = Transformation(clonedNetwork->getFunction(), clonedNetwork);
+        for (details::CNNNetworkIterator itLayer{clonedNetwork.get()}; itLayer != details::CNNNetworkIterator(); itLayer++) {
+            try {
+                mkldnn::engine eng(mkldnn::engine(mkldnn::engine::kind::cpu, 0));
+                // if we can create and have not thrown exception, then layer is supported
+                std::unique_ptr <MKLDNNNode>(MKLDNNNode::CreateNode(*itLayer, eng, extensionManager, fake_w_cache));
+                for (auto&& fusedLayerName : ngraph::getFusedNamesVector((*itLayer)->getNode())) {
+                    if (contains(originalOps, fusedLayerName)) {
+                        res.supportedLayersMap.emplace(fusedLayerName, GetName());
```

Comment:
Can you explain what the purpose of QueryNetworkResult?
Original ngraph operation before conversion can be decomposed to multiple operations and the same fusedLayerName may appear in multiple nodes. Is it expected?

Reply:
Due to the legacy the most valuble QueryNetworkResult member is supportedLayersMap field. It contains maping of layer names to plugin names. QueryNetwork takes "Network" Operation as input. So QueryNetworkResult should contain just names from this operation set on output. No mater what intermediat transformation can be. The layer from "Network"  operation set is supported if it is used to form valid operation set for Plugin.